### PR TITLE
Feature/185 feature selection

### LINKED ISSUE #185: Feature selection module
**What is your feature request related to?**
* Module request

**Please describe the module/problem**

A module to select features from a given feature space. Remove redundant features, keep features with high information value.

Possible approaches: filter, wrapper, hybrid
Possible selection methods: threshold, k-best

**Describe the solution you'd like**
Parameters: selection method, k
Input: Time series of candidate features
Output: Time series of selected features, weights of all candidate features

**Possible alternatives**
[sklearn's feature selection](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.feature_selection)

**What type of Feature Request**
* Module request

### PR DESCRIPTION
**Pull request by SMEISEN**
## Description
closes #185 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos)
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Did you update the CHANGELOG

## PR review
Anyone in the community is free to review the PR once the tests have passed.

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones to the PR so it can be classified

### CODE REVIEW

File: pywatts/modules/wrappers/sklearn_wrapper.py [hunk 1]
```diff
@@ -105,17 +105,19 @@ def transform(self, **kwargs: xr.DataArray) -> xr.DataArray:
         :return: the transformed output
         """
         x_np = self._dataset_to_sklearn_input(kwargs)
+        targets = self.targets
```
Comment (SMEISEN) [inline pywatts/modules/wrappers/sklearn_wrapper.py:109]: targets -> target names

File: pywatts/modules/wrappers/sklearn_wrapper.py [hunk 2]
```diff
@@ -105,17 +105,19 @@ def transform(self, **kwargs: xr.DataArray) -> xr.DataArray:
...
         else:
             raise KindOfTransformDoesNotExistException(
                 f"The sklearn-module in {self.name} does not have a predict or transform method",
                 KindOfTransform.PREDICT_TRANSFORM)
 
-        return self._sklearn_output_to_dataset(kwargs, prediction, self.targets)
+        return self._sklearn_output_to_dataset(kwargs, prediction, targets)
 
     def inverse_transform(self, **kwargs: xr.DataArray) -> xr.DataArray:
         """
```
Comment (SMEISEN) [inline pywatts/modules/wrappers/sklearn_wrapper.py:123]: needs also be implemented for inverse_transform

Comment (benHeid) [reply pywatts/modules/wrappers/sklearn_wrapper.py:123]: The selector Mixin has inverse transform methods. So please test if this is necessary.

File: pywatts/modules/wrappers/sklearn_wrapper.py [hunk 3]
```diff
@@ -105,17 +105,19 @@ def transform(self, **kwargs: xr.DataArray) -> xr.DataArray:
         :return: the transformed output
         """
         x_np = self._dataset_to_sklearn_input(kwargs)
+        target_names = self.targets
 
         if isinstance(self.module, TransformerMixin):
```
Comment (benHeid) [inline pywatts/modules/wrappers/sklearn_wrapper.py:110]: Please change this if statement to an elif statement and add before an if statement checking if self.module is a selector mixin. If this is the case repeat line 111 also in this if clause and set target_names=[].

File: pywatts/modules/wrappers/sklearn_wrapper.py [hunk 4]
```diff
@@ -105,17 +105,19 @@ def transform(self, **kwargs: xr.DataArray) -> xr.DataArray:
...
             prediction = self.module.transform(x_np)
+            target_names = []  # output of transformer must not match the shape of the target
```
Comment (benHeid) [inline pywatts/modules/wrappers/sklearn_wrapper.py:112]: Please remove this line and change as suggested in line 110

**Review by benHeid — CHANGES_REQUESTED** [review]
Thank you for implementing. Please consider the comments in the review.

**Review by benHeid — APPROVED** [review]

**Review by benHeid — APPROVED** [review]
